public/app/features/expressions/components/Threshold.tsx

```diff
@@ -0,0 +1,148 @@
+import { css } from '@emotion/css';
+import React, { FC, FormEvent } from 'react';
+
+import { GrafanaTheme2, SelectableValue } from '@grafana/data';
+import { ButtonSelect, InlineField, InlineFieldRow, Input, Select, useStyles2 } from '@grafana/ui';
+import { EvalFunction } from 'app/features/alerting/state/alertDef';
+
+import { ClassicCondition, ExpressionQuery, thresholdFunctions } from '../types';
+
+interface Props {
+  labelWidth: number;
+  refIds: Array<SelectableValue<string>>;
+  query: ExpressionQuery;
+  onChange: (query: ExpressionQuery) => void;
+}
+
+const defaultThresholdFunction = EvalFunction.IsAbove;
+
+export const Threshold: FC<Props> = ({ labelWidth, onChange, refIds, query }) => {
+  const styles = useStyles2(getStyles);
+
+  const defaultEvaluator: ClassicCondition = {
```

Comment:
So, we keep it as it was a classic condition, but on the backend, we evaluate it as a Math expression?

Reply:
The typing information is _very_ confusing and really needs an overhaul – it's typed as a `ClassicExpression` but that's actually a catch-all type definition for _all_ expressions (`resample`, `reduce`, etc). 

I decided not to start refactoring that in this PR as it would probably make the changes harder to review.